### PR TITLE
Implement first level test suites

### PR DESCRIPTION
**Pull request by pftom**
I have divide this front-end test into three level: [three-level](https://github.com/tutureproject/renderer/issues/16). And now, I'm implement the first-level test suites.... 😀

* Add jest test config in package.json
* Add skipLibCheck in tsconfig.json for jest test skip `libCheck`.
* Add jest test ignore config `testConfigs/empty-module.js`
* Add all components snapshot test
* Extract some common data logic in `utils/ `, `src/components/tests/utils/`
* Extract some common types in `src/types/index.js`
* Add test-needed data such as `src/components/tests/utils/tuture.yml`

### CODE REVIEW

**Comment by pftom** [comment]
Travis-CI locate build error is so difficult = =

**Comment by pftom** [comment]
I'd better add a config file called `webpack.test.config.js` for test env init.

File: src/components/tests/Content.test.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+
+import Content from '../Content';
+import tutureUtilities from '../../utils/index';
+import { Step } from '../../types/index';
+import { diff } from './utils/hunData';
```
Comment (mRcfps) [inline src/components/tests/Content.test.tsx]: Should `hunData` be `hunkData`?

File: src/components/tests/App.test.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+
+import App from '../App';
+
+test('Test App render successfully', () => {
```
Comment (mRcfps) [inline src/components/tests/App.test.tsx]: I think test description should leave the word **Test** out, like this:

```javascript
test('App render successfully', () => {});
```

File: src/components/tests/Content.test.tsx (hunk 1 above)
Comment (pftom) [reply src/components/tests/Content.test.tsx]: Nice job ! Careful boy !

File: src/components/tests/App.test.tsx (hunk 2 above)
Comment (pftom) [reply src/components/tests/App.test.tsx]: Good idea! I'll try it !
